django/test/client.py

```diff
@@ -421,93 +425,126 @@ def _get_path(self, parsed):
         # Refs comment in `get_bytes_from_wsgi()`.
         return path.decode("iso-8859-1")
 
-    def get(self, path, data=None, secure=False, **extra):
+    @staticmethod
+    def _transform_headers(headers):
+        return {
+            HttpHeaders.to_header_name(header_name): value
+            for header_name, value in headers.items()
+        }
+
+    def get(self, path, data=None, secure=False, headers=None, **extra):
         """Construct a GET request."""
         data = {} if data is None else data
         return self.generic(
             "GET",
             path,
             secure=secure,
+            headers=headers,
             **{
                 "QUERY_STRING": urlencode(data, doseq=True),
                 **extra,
             },
         )
 
     def post(
-        self, path, data=None, content_type=MULTIPART_CONTENT, secure=False, **extra
+        self,
+        path,
+        data=None,
+        content_type=MULTIPART_CONTENT,
+        secure=False,
+        headers=None,
```

Comment:
Since this is a new arg, we can enforce kwarg-only on it:

```suggestion
        *,
        headers=None,
```

(in all places)

I think this is a good idea to avoid overly long positional argument lists